crates/cli/src/lib.rs

```diff
@@ -567,22 +580,50 @@ pub fn test(matches: &ArgMatches, triple: Triple) -> io::Result<i32> {
         // running tests altogether!
         Ok(2)
     } else {
-        let failed_color = if failed_count == 0 {
-            ANSI_STYLE_CODES.green
-        } else {
-            ANSI_STYLE_CODES.red
-        };
+        let show_module_label = results_by_module.len() > 1;
+        for (module_id, failed_count, passed_count, duration) in results_by_module {
+            print_test_results(
+                module_id,
+                show_module_label,
+                failed_count,
+                passed_count,
+                duration,
+                interns,
+            );
+        }
 
-        let passed_color = ANSI_STYLE_CODES.green;
+        Ok((total_failed_count > 0) as i32)
+    }
+}
 
-        let reset = ANSI_STYLE_CODES.reset;
+fn print_test_results(
+    module_id: ModuleId,
+    show_module_label: bool,
+    failed_count: usize,
+    passed_count: usize,
+    duration: Duration,
+    interns: &Interns,
+) {
+    let failed_color = if failed_count == 0 {
+        ANSI_STYLE_CODES.green
+    } else {
+        ANSI_STYLE_CODES.red
+    };
 
+    let passed_color = ANSI_STYLE_CODES.green;
+    let reset = ANSI_STYLE_CODES.reset;
+
+    if show_module_label {
+        let module_name = module_id.to_ident_str(interns);
+        println!(
+            "\n{module_name}.roc:\n    {failed_color}{failed_count}{reset} failed and {passed_color}{passed_count}{reset} passed in {} ms.\n",
```

Comment:
Is there a more proper way to get the name of the source file here?

Reply:
One (possibly dirty) way to get the module's file path is to make use of `expectations`, 
which is a `VecMap<ModuleId, Expectations>`.

Given a `module_id` you can then get to its `Expectations` and thus to the `module`'s path via:
```roc
expectations.get(&module_id).unwrap().path
```

Reply:
We should probably pull out shared code from `report.rs:pretty_header_with_path` but we can do that in a follow-up PR.